app/controllers/categories_controller.rb

```diff
@@ -0,0 +1,76 @@
+class CategoriesController < ApplicationController
+  before_action :set_user
+  before_action :set_category, only: %i[show edit update destroy]
+
+  # GET /categories or /categories.json
+  def index
+    @categories = @user.categories
+  end
+
+  # GET /categories/1 or /categories/1.json
+  def show
+    @expenses = @category.expenses.order(created_at: :desc)
+  end
+
+  # GET /categories/new
+  def new
+    @category = @user.categories.build
+  end
+
+  # GET /categories/1/edit
+  def edit; end
```

Comment:
[ OPTIONAL] 
You can consider remvoving actions not used from your controllers. This makes your codebase clean and professional.